Mathlib/Analysis/Normed/Ring/Basic.lean

```diff
@@ -657,6 +657,112 @@ instance RingHomIsometric.ids : RingHomIsometric (RingHom.id R₁) :=
 
 end RingHomIsometric
 
+section NormMulClass
+
+/-- A mixin class for strict multiplicativity of the norm, `‖a * b‖ = ‖a‖ * ‖b‖` (rather than
+`≤` as in the definition of `NormedRing`). Many `NormedRing`s satisfy this stronger property,
+including all `NormedDivisionRing`s and `NormedField`s. -/
+class NormMulClass (α : Type*) [Norm α] [Mul α] : Prop where
+  /-- The norm is multiplicative. -/
+  protected norm_mul : ∀ (a b : α), ‖a * b‖ = ‖a‖ * ‖b‖
+
+@[simp] lemma norm_mul [Norm α] [Mul α] [NormMulClass α] (a b : α) :
+    ‖a * b‖ = ‖a‖ * ‖b‖ :=
+  NormMulClass.norm_mul a b
+
+section SeminormedAddCommGroup
+
+variable [SeminormedAddCommGroup α] [Mul α] [NormMulClass α] (a b : α)
+
+@[simp] lemma nnnorm_mul : ‖a * b‖₊ = ‖a‖₊ * ‖b‖₊ := NNReal.eq <| norm_mul a b
+
+@[simp] lemma enorm_mul : ‖a * b‖ₑ = ‖a‖ₑ * ‖b‖ₑ := by simp [enorm]
+
+end SeminormedAddCommGroup
+
+section NormedAddCommGroup
+
+variable [NormedAddCommGroup α] [MulOneClass α] [NormMulClass α] [Nontrivial α]
+
+/-- Deduce `NormOneClass` from `NormMulClass` under a suitable nontriviality hypothesis. -/
+instance NormMulClass.toNormOneClass : NormOneClass α := by
+  constructor
+  obtain ⟨u, hu⟩ := exists_ne (0 : α)
+  simpa [mul_eq_left₀ (norm_ne_zero_iff.mpr hu)] using (norm_mul u 1).symm
```

Comment:
I worry this instance and the `isDomain` one below might be a performance drag; could you split them (together) to a separate PR so that we can benchmark them in isolation?